app/models/concerns/ubiquity/works_virtual_fields.rb

```diff
@@ -109,46 +109,59 @@ def save_alternate_identifier
       end
 
       def save_date_published
-        date ||= date_published if date_published.present? && date_published.class == String
-        new_value = populate_date_field(date, date_published)
-        self.date_published_group ||= new_value if new_value.present? && date_published_group.blank?
-        self.date_published = transform_date_group(date_published_group.first) if date_published_group
+        save_single_date('date_published')
       end
 
       def save_date_accepted
-        date ||= date_accepted if date_accepted.present? && date_accepted.class == String
-        new_value = populate_date_field(date, date_accepted)
-        self.date_accepted_group ||= new_value if new_value.present? && date_accepted_group.blank?
-        self.date_accepted = transform_date_group(date_accepted_group.first) if date_accepted_group
+        save_single_date('date_accepted')
       end
 
       def save_date_submitted
-        date ||= date_submitted if date_submitted.present? && date_submitted.class == String
-        new_value = populate_date_field(date, date_accepted)
-        self.date_submitted_group ||= new_value if new_value.present? && date_submitted_group.blank?
-        self.date_submitted = transform_date_group(date_submitted_group.first) if date_submitted_group
+        save_single_date('date_submitted')
+      end
+
+      def save_single_date(value_name)
+        value = public_send(value_name)
+        # We have value and that value is a string
+        new_value = populate_date_field(value, value_name) if value.present? && value.class == String
```

Comment:
Reading  [`#populate_date_field`](https://github.com/scientist-softserv/britishlibrary/blob/94b2fb18dde9380873c5a84d608e782229d6a323/app/models/concerns/ubiquity/all_models_virtual_fields.rb#L137-L141) this is a good catch.  The parameters in the previous implementation were two values.  In this implementation, you're matching the expected parameters (e.g. value and field_name).

Good catch!

Reply:
Thanks